pkg/datapath/sockets/sockets.go

```diff
@@ -126,6 +137,107 @@ func Destroy(logger *slog.Logger, filter SocketFilter) error {
 	return nil
 }
 
+type BPFSocketDestroyer struct {
+	destroyMu lock.Mutex
+	prog      *ebpf.Program
+	filterMap *socktermfilter.Map
+	Logger    *slog.Logger
+}
+
+// Destroy destroys sockets matching the passed filter parameters using a BPF
+// socket iterator and the cil_sock_udp_destroy program.
+//
+// Supported families in the filter: syscall.AF_INET, syscall.AF_INET6
+// Supported protocols in the filter: unix.IPPROTO_UDP
+func (sd *BPFSocketDestroyer) Destroy(f SocketFilter) error {
+	if f.Family != syscall.AF_INET && f.Family != syscall.AF_INET6 {
+		return fmt.Errorf("unsupported family for socket destroy: %d", f.Family)
+	}
+	if f.Protocol != unix.IPPROTO_UDP {
+		return fmt.Errorf("unsupported protocol for socket destroy: %d", f.Protocol)
+	}
+
+	sd.destroyMu.Lock()
+	defer sd.destroyMu.Unlock()
+
+	if sd.prog == nil {
+		prog, filterMap, err := loader.LoadSockTerm("")
```

Comment:
This is something that should typically happen in a `NewSockDestroyer()`. Attempting to load the prog could be a good way of probing kernel support for the kfunc and could obviate a separate probing mechanism.

Reply:
Yeah, sure this would be cleaner. I will need to update the test code, I think, but that shouldn't be a big problem.